pkg/handler/handler.go

```diff
@@ -96,26 +100,82 @@ type ReadmeUtgaveFilter struct {
 	First  *int32
 }
 
-func (r *resolver) ReadmeUtgaver(filter *ReadmeUtgaveFilter) ([]ReadmeUtgave, error) {
-	if filter == nil {
-		return r.readmes, nil
-	}
-	var filteredReadmes []ReadmeUtgave
-	for _, r := range r.readmes {
-		if filter.Year != nil && *filter.Year != r.Year {
-			continue
+func getLink(name string) string {
+	return fmt.Sprintf("https://storage.googleapis.com/%v/%v", bucketName, name)
+}
+
+func getReadmes(ctx context.Context, client storage.Client, query string) ([]ReadmeUtgave, error) {
+	pdfQuery := &storage.Query{Prefix: "pdf/" + query}
+	imageQuery := &storage.Query{Prefix: "images/" + query}
+
+	re := regexp.MustCompile(`(?P<Year>\d{4})-(?P<Utgave>\d{2})`)
+
+	bkt := client.Bucket(bucketName)
+	pdfIt := bkt.Objects(ctx, pdfQuery)
+	imageIt := bkt.Objects(ctx, imageQuery)
+	var err error
+	var readmes  []ReadmeUtgave
+	for {
+		pdfAttrs, err := pdfIt.Next()
+		if err == iterator.Done {
```

Comment:
What if `err != nil && err != iterator.Done`, nullptr dereference?

Reply:
Hmm yes, i should probably store the errors in different variables

Reply:
ah, or just initialize as `nil`